sdk/communication/azure-communication-phonenumbers/src/main/java/com/azure/communication/phonenumbers/PhoneNumbersAsyncClient.java

```diff
@@ -129,12 +144,100 @@
      *
      * @param countryCode The ISO 3166-2 country code.
      * @param searchRequest {@link PhoneNumberSearchRequest} specifying the search request
+     * until it gets a result from the server
      * @return A {@link PollerFlux} object with the reservation result
      */
     @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
-    public PollerFlux<PollResult<PhoneNumberSearchResult>, PhoneNumberSearchResult> beginSearchAvailablePhoneNumbers(
+    public PollerFlux<PhoneNumberOperation, PhoneNumberSearchResult> beginSearchAvailablePhoneNumbers(
         String countryCode, PhoneNumberSearchRequest searchRequest) {
-        return null;
+        return beginSearchAvailablePhoneNumbers(countryCode, searchRequest, null);
+    }
+
+    PollerFlux<PhoneNumberOperation, PhoneNumberSearchResult> beginSearchAvailablePhoneNumbers(
+        String countryCode, PhoneNumberSearchRequest searchRequest, Context context) {
+        try {
+            Objects.requireNonNull(countryCode, "'countryCode' cannot be null.");
+            Objects.requireNonNull(searchRequest, "'searchRequest' cannot be null.");
+
+            return new PollerFlux<>(defaultPollInterval,
+                searchAvailableNumbersInitOperation(countryCode, searchRequest, context),
+                pollOperation(),
+                cancelOperation(),
+                searchAvailableNumbersFetchFinalResultOperation());
+
+        } catch (RuntimeException ex) {
+            return PollerFlux.error(ex);
+        }
+    }
+
+    private Function<PollingContext<PhoneNumberOperation>, Mono<PhoneNumberOperation>>
+        searchAvailableNumbersInitOperation(String countryCode, PhoneNumberSearchRequest searchRequest, Context context) {
+        return (pollingContext) -> {
+            return withContext(contextValue -> {
+                if (context != null) {
+                    contextValue = context;
+                }
+                return client.searchAvailablePhoneNumbersWithResponseAsync(countryCode, searchRequest, contextValue)
+                .flatMap((PhoneNumbersSearchAvailablePhoneNumbersResponse response) -> {
+                    pollingContext.setData("operationId", parseIdFromUrl(response.getDeserializedHeaders().getOperationLocation()));
+                    pollingContext.setData("searchId", parseIdFromUrl(response.getDeserializedHeaders().getLocation()));
+                    return client.getOperationAsync(pollingContext.getData("operationId"))
+                    .flatMap((PhoneNumberOperation result) -> {
+                        return Mono.just(result);
```

Comment:
Is there a reason you're using this flatMap operation? It just retuns the result as-is without transformation.

Reply:
That's a good point, I will update. thank you!